components/organisation-unit-tree/src/organisation-unit-tree/use-root-org-data/use-root-org-data.js

```diff
@@ -29,37 +29,18 @@ export const createRootQuery = ids =>
 export const useRootOrgData = (ids, { isUserDataViewFallback } = {}) => {
     const query = createRootQuery(ids)
     const variables = { isUserDataViewFallback }
-    const [state, setState] = useState({
-        loading: true,
-        error: null,
-        data: null,
-    })
-
-    const { refetch: queryRefetch } = useDataQuery(query, {
+    const { loading, error, data, refetch } = useDataQuery(query, {
         variables,
-        onComplete: queryData =>
-            setState({
-                ...state,
-                data: queryData ? patchMissingDisplayName(queryData) : {},
-                loading: false,
-            }),
-        onError: queryError =>
-            setState({
-                ...state,
-                error: queryError,
-                loading: false,
-            }),
     })
 
-    const refetch = useCallback(() => {
-        setState({ ...state, data: null, error: null, loading: true })
-        queryRefetch()
-    }, [queryRefetch, setState])
+    const patchedData = useMemo(() => {
+        return data ? patchMissingDisplayName(data) : data
+    }, [data])
 
     return {
-        loading: state.loading,
-        error: state.error,
-        data: state.data,
+        loading,
+        error: error || null,
+        data: patchedData || null,
         refetch,
     }
```

Comment:
Eventually we should refactor this a little more in depth, so that the hook just passes around the default values that the app-runtime uses as well. Might be feasible to do that in this PR, didn't look into it (also goes for other areas where I'm sticking closely to the existing API of the refactored components for safety).